src/index.rs

```diff
@@ -984,4 +1003,198 @@ mod tests {
+
+                for (add_path, &add_id) in one_index_update.added.iter() {
+                    all_added.insert(add_path.clone(), add_id.clone());
+                }
+
+                for delete_id in one_index_update.deleted {
+                    all_deleted.insert(delete_id);
+                }
+            }
+
+            let curr_paths: HashMap<CanonicalPathBuf, DirEntry> =
+                discover_paths(path.clone());
+            let curr_entries = scan_entries(curr_paths);
+
+            for (cur_path, cur_entry) in curr_entries {
+                let one_index_update = actual
+                    .update_one(&cur_path.clone(), cur_entry.id)
+                    .expect("Should update one resource correctly");
+
+                for (add_path, &add_id) in one_index_update.added.iter() {
+                    all_added.insert(add_path.clone(), add_id.clone());
+                }
+
+                for delete_id in one_index_update.deleted {
+                    all_deleted.insert(delete_id);
+                }
+            }
+
+            for (key, entry) in &actual.path2id {
+                println!(
+                    "===😂😂========update_after_entry Resource ID: {}",
+                    entry.id
+                );
+            }
+
+            println!(
+                "===✔✔==added=:{}   deleted={}",
+                all_added.len().to_string(),
+                all_deleted.len().to_string()
+            );
+        })
+    }
+
+    #[test]
+    fn should_correctly_update_all_compare_update_one_resource() {
+        run_test_and_clean_up(|path| {
+            create_file_at(path.clone(), Some(FILE_SIZE_2), Some(FILE_NAME_2));
+            let mut actual = ResourceIndex::build(path.clone());
+            create_file_at(path.clone(), Some(FILE_SIZE_1), Some(FILE_NAME_1));
+
+            let updated_using_update_all_result = actual.clone().update_all();
+
+            let mut all_deleted: HashSet<ResourceId> = HashSet::new();
+            let mut all_added: HashMap<CanonicalPathBuf, ResourceId> =
+                HashMap::new();
+
+            let curr_paths: HashMap<CanonicalPathBuf, DirEntry> =
+                discover_paths(path.clone());
+            let curr_entries = scan_entries(curr_paths);
+
+            for (cur_path, cur_entry) in curr_entries {
+                let one_index_update = actual
+                    .update_one(&cur_path.clone(), cur_entry.id)
+                    .expect("Should update one resource correctly");
+
+                for (add_path, &add_id) in one_index_update.added.iter() {
+                    all_added.insert(add_path.clone(), add_id.clone());
+                }
+
+                for delete_id in one_index_update.deleted {
+                    all_deleted.insert(delete_id);
+                }
+            }
+
+            let updated_using_update_one = IndexUpdate {
+                deleted: all_deleted,
+                added: all_added,
+            };
```

Comment:
It's a good idea to compare `IndexUpdate` structures, however we should focus first on `Index` instances themselves.

Reply:
Something like this is easier to start with:
```
    #[test]
    fn sequence_of_update_one_calls_should_imitate_update_all() {
        run_test_and_clean_up(|path| {
            create_file_at(path.clone(), Some(FILE_SIZE_2), Some(FILE_NAME_2));
            let index_initial = ResourceIndex::build(path.clone());
            create_file_at(path.clone(), Some(FILE_SIZE_1), Some(FILE_NAME_1));

            let mut index_full = index_initial.clone();
            index_full.update_all()
                .expect("Should update the index correctly");

            let curr_paths: HashMap<CanonicalPathBuf, DirEntry> =
                discover_paths(path.clone());
            let curr_entries = scan_entries(curr_paths);

            let mut index_by_one = index_initial.clone();

            // only 2 iterations at this moment
            for (cur_path, cur_entry) in curr_entries {
                index_by_one
                    .update_one(&cur_path.clone(), cur_entry.id)
                    .expect("Should update one resource correctly");
            }

            assert_eq!(index_full, index_by_one);
        })
    }
```